src/simulator/simulator.c

```diff
@@ -130,6 +133,17 @@ simulator_add_callback(State* state, Simulator* simulator, Callback* callback) {
     simulator->callbacks[simulator->n_callbacks] = callback;
     ++(simulator->n_callbacks);
     
+    error:
+    return;
+}
+
+
+internal void
+simulator_destroy(Simulator* simulator) {
+    check(simulator != NULL, "simulator is NULL");
+    
+    thread_pool_stop(simulator->pool);
```

Comment:
Actually, can we just call this inside the thread_pool_destroy? and remove the destroy for this?

Reply:
No, but no need to keep it in the simulator, just use it as a parameter and call stop in the experiment manually